core/src/main/java/tc/oc/pgm/filters/dynamic/FilterMatchModule.java

```diff
@@ -107,9 +130,26 @@ public void onMatchLoad(MatchLoadEvent event) {
             });
   }
 
+  private void findAndCreateReactorFactories(Filter filter) {
+    if (filter instanceof ParentFilter) {
+      ((ParentFilter) filter).getChildren().forEach(this::findAndCreateReactorFactories);
+    }
+    if (filter instanceof XMLFilterReference) {
+      this.findAndCreateReactorFactories(((XMLFilterReference) filter).get());
+    }
```

Comment:
```suggestion
    if (filter instanceof XMLFilterReference) {
      this.findAndCreateReactorFactories(((XMLFilterReference) filter).get());
      return;
    }
```
i'd move this up above parent filter, if it's an xml reference just resolve it and return, no need to check for parent filter or reactor factory